plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergVariantParquet.java

```diff
@@ -0,0 +1,272 @@
+    private static final VariantMetadata EMPTY_METADATA = Variants.metadata(VariantTypeTestUtils.emptyMetadata());
+    private static final VariantObject TEST_OBJECT = (VariantObject) Variants.value(TEST_METADATA, TEST_OBJECT_BUFFER);
+    private static final VariantObject SIMILAR_OBJECT = (VariantObject) Variants.value(TEST_METADATA, SIMILAR_OBJECT_BUFFER);
+    private static final VariantObject EMPTY_OBJECT = (VariantObject) Variants.value(TEST_METADATA, EMPTY_OBJECT_BUFFER);
+
+    @Override
+    protected QueryRunner createQueryRunner()
+            throws Exception
+    {
+        warehouseLocation = Files.createTempDirectory("test_iceberg_variant_test").toFile();
+        closeAfterClass(() -> deleteRecursively(warehouseLocation.toPath(), ALLOW_INSECURE));
+        TestingIcebergJdbcServer server = closeAfterClass(new TestingIcebergJdbcServer());
+        jdbcCatalog = (JdbcCatalog) buildIcebergCatalog("tpch", ImmutableMap.<String, String>builder()
+                        .put(CATALOG_IMPL, JdbcCatalog.class.getName())
+                        .put(URI, server.getJdbcUrl())
+                        .put(PROPERTY_PREFIX + "user", USER)
+                        .put(PROPERTY_PREFIX + "password", PASSWORD)
+                        .put(PROPERTY_PREFIX + "schema-version", IcebergJdbcCatalogConfig.SchemaVersion.V1.toString())
+                        .put(WAREHOUSE_LOCATION, warehouseLocation.getAbsolutePath())
+                        .buildOrThrow(),
+                new Configuration(false));
+        return IcebergQueryRunner.builder()
+                .setIcebergProperties(
+                        ImmutableMap.<String, String>builder()
+                                .put("iceberg.file-format", PARQUET)
+                                .put("iceberg.catalog.type", "jdbc")
+                                .put("iceberg.jdbc-catalog.driver-class", "org.postgresql.Driver")
+                                .put("iceberg.jdbc-catalog.connection-url", server.getJdbcUrl())
+                                .put("iceberg.jdbc-catalog.connection-user", USER)
+                                .put("iceberg.jdbc-catalog.connection-password", PASSWORD)
+                                .put("iceberg.jdbc-catalog.catalog-name", "tpch")
+                                .put("iceberg.register-table-procedure.enabled", "true")
+                                .put("iceberg.writer-sort-buffer-size", "1MB")
+                                .put("iceberg.allowed-extra-properties", "write.metadata.delete-after-commit.enabled,write.metadata.previous-versions-max")
+                                .put("iceberg.jdbc-catalog.default-warehouse-dir", warehouseLocation.getAbsolutePath())
+                                .put("iceberg.jdbc-catalog.retryable-status-codes", "57P01,57P05")
+                                .buildOrThrow())
+                .build();
+    }
+
+    @ParameterizedTest
+    @MethodSource("variantTypeResultMappings")
```

Comment:
https://trino.io/docs/current/develop/tests.html#avoid-data-providers-and-parametric-tests